data-raw/simulate-data.R

```diff
@@ -411,27 +411,39 @@ simulate_data <- function(data, n) {
     list_cbind()
 }
 
-set.seed(123)
-simulation_definitions <- here("data-raw/simulation-definitions.csv") |>
-  read_csv(show_col_types = FALSE) |>
-  select(register_abbrev, variable_name, generator)
-
-simulation_definitions_list <- simulation_definitions |>
-  group_split(register_abbrev)
+#' Read simulation definitions from csv
+#'
+#' @param path Path to the csv file.
+#'
+#' @returns A tibble with simulation definitions.
+read_simulation_definitions_data <- function(path) {
+  path |>
+    readr::read_csv(show_col_types = FALSE) |>
+    dplyr::select("register_abbrev", "variable_name", "generator")
+}
 
-register_data <- simulation_definitions_list |>
-  map(\(data) simulate_data(data, n = 1000)) |>
-  map(insert_specific_atc) |>
-  map(add_fake_drug_name) |>
-  map(insert_false_drug_names) |>
-  map(insert_false_metformin) |>
-  map(insert_analysiscode) |>
-  # add the register abbreviations as a name to the list
-  set_names(
-    map_chr(
-      simulation_definitions_list,
-      \(data) unique(data$register_abbrev)
+set.seed(123)
```

Comment:
create_simulated_data

Reply:
No, not if it is outside the function that needs the set seed.

Reply:
But isn't it bad practice to set a seed inside a function? Maybe not?

Reply:
(I've moved it inside the function now)

Reply:
Or set the seed in the targets pipeline: https://books.ropensci.org/targets/random.html#reproducibility

Reply:
Done!